codec/depinject.go

```diff
@@ -0,0 +1,43 @@
+package codec
+
+import (
+	"github.com/cosmos/gogoproto/proto"
+
+	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/legacy"
+	"cosmossdk.io/x/tx/signing"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+func ProvideInterfaceRegistry(
+	addressCodec address.Codec,
+	validatorAddressCodec address.ValidatorAddressCodec,
+	customGetSigners []signing.CustomGetSigner,
+) (types.InterfaceRegistry, error) {
+	signingOptions := signing.Options{
+		AddressCodec:          addressCodec,
+		ValidatorAddressCodec: validatorAddressCodec,
+	}
+	for _, signer := range customGetSigners {
+		signingOptions.DefineCustomGetSigners(signer.MsgType, signer.Fn)
+	}
+
+	interfaceRegistry, err := types.NewInterfaceRegistryWithOptions(types.InterfaceRegistryOptions{
+		ProtoFiles:     proto.HybridResolver,
+		SigningOptions: signingOptions,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := interfaceRegistry.SigningContext().Validate(); err != nil {
+		return nil, err
+	}
+
+	return interfaceRegistry, nil
+}
+
+func ProvideCodecs(interfaceRegistry types.InterfaceRegistry) (legacy.Amino, *ProtoCodec) {
+	return NewLegacyAmino(), NewProtoCodec(interfaceRegistry)
```

Comment:
Would it make sense to have a separate Provide function for the legacy codec? Making it easier to deprecate later. Not a blocker tho, just asking for opinions

Reply:
This makes sense to me, breaking it in two providers would be great.